### ISSUE TITLE
2fa Gitea

### ISSUE THREAD

**Issue by 12nick12** [woodpecker-ci/woodpecker#195]
Should this work with 2fa and gitea? When I try to use my gitea account I get the following error

time="2021-05-03T16:00:47Z" level=error msg="cannot authenticate user. basic auth required"

**Comment by 6543** [comment]
currently woodpecker login via BasicAuth, so 2fa+gitea is not supported until woodpecker move to Oauth2 Login ...

**Comment by 12nick12** [comment]
Ah ok, How do I go about creating the basic auth users? Or would 2fa need to be disable in Gitea?

**Comment by 6543** [comment]
a quick fix would be to disable 2fa, but this is not realy a good idear :/
I think I can have a look at let woodpecker accept token, this way it would be possible ...

**Comment by 6543** [comment]
-> #223 
```go
// If username and password is set, Basic Auth is used.
// If only username is set, it is used as token.
```

**Comment by 6543** [comment]
@12nick12 can you test this pull, if it works for you?

**Comment by 12nick12** [comment]
I ended up switching to metroline. If I get bored I'll give it a go. Thank you for your work :-)

**Comment by 6543** [comment]
closed by #226
